src/main/scala/SBTS3Resolver.scala

```diff
@@ -79,7 +82,22 @@ object SbtS3Resolver extends AutoPlugin {
 
       // convenience method, to use normal bucket addresses with `at`
       // without this resolver: "foo" at s3("maven.bucket.com").toHttps(s3region.value)
-      def toHttps(region: String): String = s"""https://s3-${region}.amazonaws.com/${url.stripPrefix("s3://")}"""
+      def toHttps(region: String): String = {
+        val bucketPath = url.stripPrefix("s3://")
+        val euCentral = Region.EU_Frankfurt.toString
+
+        //Correcting for fact that region enum resolves to null when provided US_STANDARD (us-east-1)
+        val correctedRegion = if(region == null) "us-east-1" else region
+
+        /*
+        Constructing host name based on region name. Frankfurt host name must be constructed using s3.{region}
+        for java sdk to sign its request using the v4 signature method (Frankfurt only supports v4 signing)
+         */
+        region match {
+          case euCentral => s"""https://s3.${correctedRegion}.amazonaws.com/${bucketPath}"""
+          case _ => s"""https://s3-${correctedRegion}.amazonaws.com/${bucketPath}"""
+        }
```

Comment:
Here's a solution:

```scala
import com.amazonaws.services.s3._, model._

Region.EU_Frankfurt.toAWSRegion.getServiceEndpoint(AmazonS3.ENDPOINT_PREFIX)
Region.US_Standard.toAWSRegion.getServiceEndpoint(AmazonS3.ENDPOINT_PREFIX)
```

It handles the different prefixes (`.`/`-` and `.cn`) and also works for `US_Standard` (for some reason `Region.US_Standard.toAWSRegion` doesn't throw a NPE).
I guess, I'll also change the type of the parameter to `com.amazonaws.regions.Region` to avoid these problems.